src/libraries/Microsoft.Extensions.Options/tests/Microsoft.Extensions.Options.Tests/OptionsValidationBuilderTests.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Extensions.Options.Tests
+{
+    public class OptionsValidationBuilderTests
+    {
+        [Fact]
+        public void ValidateBuilder()
```

Comment:
Nit: would recommend factoring individual groups of assertions into separate unit tests for easier tracking of potential future regressions.

That being said, the type seems simple enough so maybe not too important.